client/my-sites/upgrades/index.js

```diff
@@ -284,10 +285,19 @@ module.exports = function() {
 		);
 
 		page(
-			'/checkout/:domain/:plan_name?',
+			'/checkout/:domain/:planName?',
 			adTracking.retarget,
 			controller.siteSelection,
 			upgradesController.checkout
 		);
+
+		page(
+			'/start-trial/:planName/:domain',
```

Comment:
I feel I already mentioned it but I'm pretty sure we should use `:site` instead of `:domain` in our routes.